earth2mip/forecasts.py

```diff
@@ -70,6 +70,25 @@ async def __getitem__(self, i: int):
             yield x
 
 
+class select_channels(Forecast):
+    def __init__(self, forecast: Forecast, channel_names: list[str]):
+        self.forecast = forecast
+        self._channel_names = channel_names
+        self._index = [self.forecast.channel_names.index(x) for x in self.channel_names]
+
+    @property
+    def channel_names(self):
+        return self._channel_names
+
+    @property
+    def grid(self):
+        return self.forecast.grid
+    
+    async def __getitem__(self, i: int):
+        async for x in self.forecast[i]:
+            yield x[:, self._index]
+
+
```

Comment:
I don't understand the motivation for making this an object, or an object of type `Forecast`. What is a `Forecast` and why should a function that selects channels be a type of it? This seems like it should be more of a utility function rather than a class.

Reply:
I explain the motivation for the `Forecast` API here: https://nvidia.github.io/earth2mip/userguide/concepts.html#forecast

> Many scoring algorithms are most easily expressed as operations over 2D array of states that we call a Forecast Array. The rows of this array correspond to initial times, and the columns to lead times. The size of this array may be unbounded. For example, computing a lead time dependent metrics, such as RMSE corresponds to averaging the square difference of Forecast Arrays of observations and forecasts, and then averaging over the row dimension. This is defined by the [earth2mip.forecasts.Forecast](https://nvidia.github.io/earth2mip/modules/index.html#earth2mip.forecasts.Forecast) interface. Compared to a TimeLoop, a Forecast encapsulates any time handling and initialization logic. One advantage is that an archive of forecasts on disk can be represented as a Forecast (see [earth2mip.forecasts.XarrayForecast](https://nvidia.github.io/earth2mip/modules/index.html#earth2mip.forecasts.XarrayForecast)). This allows using the same code to score both static and streaming forecasts.

Reply:
I chose to use lower case since this is a `Callable[[Forecast,...], Forecast` that I intended to be used like a function. See https://github.com/NVIDIA/earth2mip/pull/154#discussion_r1440909737 for more info.

I could make it a class `ChannelSubsetForecast` and write a `select_channels(forecast: Forecast, ...) -> ChannelSubsetForecast`, but the effect is the same and this is less work.